A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_PerformancePage.js

```diff
@@ -48,158 +48,298 @@ class CDUPerformancePage {
+                } else {
+                    if (mcdu.trySetV1Speed(value)) {
+                        CDUPerformancePage.ShowTAKEOFFPage(mcdu);
+                    }
                 }
+            };
+            vR = "{amber}___{end}";
+            if (mcdu.vRSpeed) {
+                vR = `{cyan}${("" + mcdu.vRSpeed).padEnd(3)}{end}`;
+            }
+            mcdu.onLeftInput[1] = (value) => {
+                if (value === FMCMainDisplay.clrValue) {
+                    mcdu.vRSpeed = undefined;
+                    SimVar.SetSimVarValue("L:AIRLINER_VR_SPEED", "Knots", -1);
+                    CDUPerformancePage.ShowTAKEOFFPage(mcdu);
+                } else if (value === "") {
+                    mcdu.sendDataToScratchpad(mcdu._getVRSpeed().toString());
+                    CDUPerformancePage.ShowTAKEOFFPage(mcdu);
+                } else {
+                    if (mcdu.trySetVRSpeed(value)) {
+                        CDUPerformancePage.ShowTAKEOFFPage(mcdu);
+                    }
+                }
+            };
+            v2 = "{amber}___{end}";
+            if (mcdu.v2Speed) {
+                v2 = `{cyan}${("" + mcdu.v2Speed).padEnd(3)}{end}`;
+            }
+            mcdu.onLeftInput[2] = (value) => {
+                if (value === FMCMainDisplay.clrValue) {
+                    mcdu.v2Speed = undefined;
+                    SimVar.SetSimVarValue("L:AIRLINER_V2_SPEED", "Knots", -1);
+                    CDUPerformancePage.ShowTAKEOFFPage(mcdu);
+                } else if (value === "") {
+                    mcdu.sendDataToScratchpad(mcdu._getV2Speed().toString());
+                    CDUPerformancePage.ShowTAKEOFFPage(mcdu);
+                } else {
+                    if (mcdu.trySetV2Speed(value)) {
+                        CDUPerformancePage.ShowTAKEOFFPage(mcdu);
+                    }
+                }
+            };
+        } else {
+            v1 = "\xa0\xa0\xa0";
+            vR = "\xa0\xa0\xa0";
+            v2 = "\xa0\xa0\xa0";
+            if (mcdu.v1Speed) {
+                v1 = `{green}${("" + mcdu.v1Speed).padEnd(3)}{end}`;
             }
-        };
-        let vR = "___[color]amber";
-        if (mcdu.vRSpeed) {
-            vR = mcdu.vRSpeed + "[color]cyan";
+            if (mcdu.vRSpeed) {
+                vR = `{green}${("" + mcdu.vRSpeed).padEnd(3)}{end}`;
+            }
+            if (mcdu.v2Speed) {
+                v2 = `{green}${("" + mcdu.v2Speed).padEnd(3)}{end}`;
+            }
+            mcdu.onLeftInput[0] = (value) => {
+                if (value !== "") {
+                    mcdu.showErrorMessage("NOT ALLOWED");
+                }
+            };
+            mcdu.onLeftInput[1] = (value) => {
+                if (value !== "") {
+                    mcdu.showErrorMessage("NOT ALLOWED");
+                }
+            };
+            mcdu.onLeftInput[2] = (value) => {
+                if (value !== "") {
+                    mcdu.showErrorMessage("NOT ALLOWED");
```

Comment:
`showErrorMessage` no longer exists and has been replaced with a general message handler that takes a message object as parameter.

Example of the new system:

`addNewMessage(NXSystemMessages.notAllowed);`

Reply:
Thanks, that code is quite old as you can see :-)